python-frontend/src/main/java/org/sonar/python/semantic/v2/types/Propagation.java

```diff
@@ -106,26 +54,22 @@ Set<Propagation> dependents() {
     return dependents;
   }
 
-  public abstract Name lhsName();
+  public Name lhsName() {
+    return lhsName;
+  }
 
   public abstract PythonType rhsType();
 
-  @CheckForNull
-  static PythonType currentType(Name lhsName) {
-    return Optional.ofNullable(lhsName.symbolV2())
-      .stream()
-      .map(SymbolV2::usages)
-      .flatMap(List::stream)
-      .filter(u -> !SymbolV2Utils.isDeclaration(u))
-      .map(UsageV2::tree)
-      .filter(Expression.class::isInstance)
-      .map(Expression.class::cast)
-      .findFirst()
-      .map(Expression::typeV2)
-      .orElse(null);
-  }
-
   public SymbolV2 lhsSymbol() {
     return lhsSymbol;
   }
+
+  public void addVariableDependencies(SymbolV2 dependency) {
```

Comment:
should it be named singular like `addVariableDependency`?

Reply:
it should. thx